### PR TITLE
Navbar :First Iteration

### LINKED ISSUE #52: [Site] Add the Navbar section
**Desired State of the Service Mesh Patterns site**

A `NavBar` section needs to be added to the site as per this [Figma](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs?node-id=8838%3A14491) design.

---
### Contributor Resources
This website uses React and GitHub Pages. Site content is found under the [`master` branch](https://github.com/service-mesh-books/service-mesh-patterns/tree/master/src).
- See the [Contributing to Service Mesh Patterns Website](https://github.com/service-mesh-patterns/service-mesh-patterns/#readme) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- Layer5 [Brand Kit](https://layer5.io/company/brand)
- 📑 [Service Mesh Patterns in Meshery Catalog](https://meshery.io/catalog)
- 📖 [Service Mesh Patterns Book](https://layer5.io/learn/service-mesh-books/service-mesh-patterns)
- 🎨 [Site designs in Figma](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs?node-id=5310%3A0). Fill-in the [Layer5 Community Member Form](https://layer5.io/newcomer) and join the [Layer5 Community Slack](http://slack.layer5.io) for access.
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by legendarykamal**
Signed-off-by: Kamal Nayan <geniusamansingh@gmail.com>

**Description**
This is the navbar component

This PR fixes #52

**Notes for Reviewers**
Currently it looks like this
image

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by legendarykamal** [comment]
This is under construction but any suggestions are welcome :)

**Comment by l5io** [comment]
🚀 Preview for commit 0d5163c2cca9e9a781f76dc1dba8da55a87a54fe at: https://62f253a5d3be3f3b83f4777f--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit ab24a3c7c86e5a50c29e65618b12be43ed94df94 at: https://62f2583f822c9e3ec0d1286c--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 3b8465f27521438b12b8a53f80dd1072f5c82098 at: https://62f7b202c580452cd32f30b8--service-mesh-patterns.netlify.app

**Comment by legendarykamal** [comment]
Updated ss
image

**Comment by Nikhil-Ladha** [comment]
The navbar is not full-width.

File: site/src/components/Navbar.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <path d="M73.5719 0.550903H68.867V2.81739H73.5719V0.550903Z" fill="#00B39F"/>
+      <path d="M74 8.18617V7.25231C74 6.17066 73.0904 5.29414 71.9692 5.29414H64.3761V4.67114H61.5962V7.34659H71.2201V8.18362L74 8.18617Z" fill="#00B39F"/>
+      <path d="M64.3761 3.93088V2.81739H68.0988V0.550903H61.5962V3.93088H64.3761Z" fill="#00D3A9"/>
+      <path d="M71.2201 8.91492V10.2335H67.4974V12.5H72.0839C72.5819 12.5079 73.0627 12.3179 73.4209 11.9719C73.7791 11.6259 73.9855 11.1519 73.9949 10.654V8.91492H71.2201Z" fill="#00D3A9"/>
+      <path d="M64.1531 9.03589V10.2335H66.7292V12.5H63.4167C62.9187 12.5078 62.4379 12.3179 62.0797 11.9718C61.7215 11.6258 61.5151 11.1519 61.5057 10.6539V9.03589H64.1531Z" fill="#00B39F"/>
+      </svg>
```
Comment (Nikhil-Ladha) [inline site/src/components/Navbar.js:24]: Simply add the layer5 svg file from the layer5.io repo and import that. No need to use a custom svg

File: site/src/components/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
```
Comment (Nikhil-Ladha) [inline site/src/components/Navbar.js:1]: This file should be under `sections` and not `components`
Create a folder named `Navbar` under `sections` and then create an index.js file and a `navbar.style.js` under it

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
Please use semantic HTML.
The current HTML structure is completely wrong.
Check any navbar code on how the structure should be, or layer5.io site itself

File: site/src/components/Navbar.js (hunk 2 above)
Comment (legendarykamal) [reply site/src/components/Navbar.js:1]: done :+1:

Comment (legendarykamal) [reply site/src/components/Navbar.js:1]: done :+1:

**Comment by legendarykamal** [comment]
width set to full
image

**Comment by l5io** [comment]
🚀 Preview for commit ec5d50e9e2d2d4b3d2781b06332bcbae81d52bf2 at: https://62fa223cfc1ab237dea08fc1--service-mesh-patterns.netlify.app

File: site/src/sections/Navbar/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const Navbar = () => {
+  return (
+    <NavStyleWrapper>
+      <img className="pattern" src={pattern} alt="pattern" />
+      <h2 className="text">Book Chapters</h2>
+      <h2 className="text">Patterns In-Action</h2>
+      <h2 className="text">Meet the Authors</h2>
+      <h2 className="text">About 
+      <img className="pattern" src={Layer5} alt="Layer5" />
+      </h2>
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/index.js:14]: Still incorrect HTML.
It should be of the form
```
<nav>
    <ul>
        <li></li>
        ...........
    </ul>
</nav>
```

Comment (legendarykamal) [reply site/src/sections/Navbar/index.js:14]: yes yes I will fix this by today

**Comment by l5io** [comment]
🚀 Preview for commit 328f0897c57a1fe939deb9165ba3e16099147016 at: https://6300a61444ba2e22e4d4fe19--service-mesh-patterns.netlify.app

File: site/src/sections/Navbar/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <NavStyleWrapper>
+      <nav>
+        <ul className="navbar">
+          <img className="pattern" src={pattern} alt="pattern" />
+          <li className="text">Book Chapters</li>
+          <li className="text">Patterns In-Action</li>
+          <li className="text">Meet the Authors</li>
+          <li className="text">About <img className="pattern" src={Layer5} alt="Layer5" /> </li>
+          <div className="button"><span className="label">Get it Now</span></div>
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/index.js:15]: Use a button for a button 🤷🏼‍♂️

File: site/src/sections/Navbar/navbar.style.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    .text1{
+        width: 67px;
+        height: 44px;
+        font-family: 'Open Sans';
+        font-style: normal;
+        font-weight: 400;
+        font-size: 16px;
+        line-height: 22px;
+
+        color: #3C494F;
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/navbar.style.js:71]: I don't think these classes are used anymore

File: site/src/sections/Navbar/navbar.style.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+export  const NavStyleWrapper = styled.nav`
+    .navbar{
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        align-items: center;
+        padding: 28px 184px;
+        gap: 100px;
+        padding-right: 70px;
+        position: absolute;
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/navbar.style.js:12]: Use `sticky`

File: site/src/sections/Navbar/index.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+const Navbar = () => {
+  return (
+    <NavStyleWrapper>
+      <nav>
+        <ul className="navbar">
+          <img className="pattern" src={pattern} alt="pattern" />
+          <li className="text">Book Chapters</li>
+          <li className="text">Patterns In-Action</li>
+          <li className="text">Meet the Authors</li>
+          <li className="text">About <img className="pattern" src={Layer5} alt="Layer5" /> </li>
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/index.js:14]: We don't really need a className here. All are `li` can be used directly in the CSSS as `li`

File: site/src/pages/index.js [hunk 8]
```diff
@@ -4,6 +4,7 @@ import SeoComponent from "../components/seo";
 // import Hero from "../sections/Hero";
 import SectionWrapper from "./index.style";
 import PatternsLogo from "../assets/images/brand/SVG/service-mesh-patterns-side_white.svg";
+import Navbar from "../sections/Navbar/index";
```
Comment (Nikhil-Ladha) [inline site/src/pages/index.js:7]: By default the `index` files are imported, you don't need to mention that explicitly.

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]

**Comment by Nikhil-Ladha** [comment]
The navbar is not full-width and not to the top of the page.

File: site/src/pages/index.js [hunk 9]
```diff
@@ -29,10 +30,10 @@ const IndexPage = () => {
             <h1 className="center under-construction">Coming Soon...</h1>
             <h3 className="under-construction">See <a href="https://layer5.io/books/service-mesh-patterns">Service Mesh Patterns Book</a></h3>
           </div>
-          {/* <Navbar />
-        <Hero />
+          <Navbar />
```
Comment (Nikhil-Ladha) [inline site/src/pages/index.js:33]: It should be the first component in the DOM tree.

File: site/src/sections/Navbar/navbar.style.js (hunk 6 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/navbar.style.js:12]: yep was gonna

**Comment by legendarykamal** [comment]
ok @Nikhil-Ladha I will look into and make changes by today :+1:

File: site/src/pages/index.js (hunk 9 above)
Comment (legendarykamal) [reply site/src/pages/index.js:33]: rightly pointed :eyes:

**Comment by legendarykamal** [comment]
Now It is at the top with full width 
image

File: site/src/pages/index.js (hunk 8 above)
Comment (legendarykamal) [reply site/src/pages/index.js]: done

File: site/src/sections/Navbar/index.js (hunk 7 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/index.js]: :+1:

File: site/src/sections/Navbar/navbar.style.js (hunk 5 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/navbar.style.js]: removed them

File: site/src/sections/Navbar/index.js (hunk 4 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/index.js]: :+1

File: site/src/sections/Navbar/navbar.style.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
+import styled from "styled-components";
+
+export  const NavStyleWrapper = styled.nav`
+    
+    .nav{
+        position: sticky;
+    }
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/navbar.style.js:7]: I don't think this class is used anywhere

File: site/src/sections/Navbar/navbar.style.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    
+    .navbar{
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        align-items: center;
+        padding: 40px 184px;
+        gap: 100px;
+        padding-right: 70px;
+        position: relative;
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/navbar.style.js:17]: sticky

File: site/src/sections/Navbar/navbar.style.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        font-size: 16px;
+        line-height: 22px;
+        
+        color: #3C494F;
+
+    }
+
+    .pattern{
+        // position: relative;
+        // top: 75px;
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/navbar.style.js:41]: Remove unused CSS

File: site/src/sections/Navbar/navbar.style.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    .nav{
+        position: sticky;
+    }
+    
+    .navbar{
+        display: flex;
+        flex-direction: row;
+        justify-content: space-between;
+        align-items: center;
+        padding: 40px 184px;
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/navbar.style.js:14]: Please use `rem` everywhere

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
There seems to be a merge conflict, please resolve that

**Comment by l5io** [comment]
🚀 Preview for commit 06246212defd10d0639319b298e07e44583f0ba9 at: https://6309e4c3bb46d553a86e264b--service-mesh-patterns.netlify.app

File: site/src/sections/Navbar/navbar.style.js (hunk 13 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/navbar.style.js]: Done have replaced ```px``` with ```rem```

File: site/src/sections/Navbar/navbar.style.js (hunk 10 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/navbar.style.js]: removed it :+1:

File: site/src/pages/index.js [hunk 14]
```diff
@@ -21,19 +22,17 @@ const IndexPage = () => {
       <SeoComponent title="Service Mesh Patterns" description="Service Mesh Patterns - Expect more from your infrastructure"
         schemaMarkup={schema} />
       <Layout>
+        <Navbar />
         <SectionWrapper>
-          <Hero />
```
Comment (debo19) [inline site/src/pages/index.js:25]: Why have you removed the banner?

File: site/src/sections/Navbar/index.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+const Navbar = () => {
+  return (
+    <NavStyleWrapper>
+      <nav>
+        <ul className="navbar">
+          <img className="pattern" src={pattern} alt="pattern" />
+          <li>Book Chapters</li>
+          <li>Patterns In Action</li>
+          <li>Meet the Authors</li>
+          <li>About <img className="pattern" src={Layer5} alt="Layer5" /> </li>
```
Comment (debo19) [inline site/src/sections/Navbar/index.js:15]: Layer5 image should be aligned with the text above (i.e. About)

**Comment by legendarykamal** [comment]
Yes @debo19 addition of a hamberger for it remains, I will add it soon

File: site/src/sections/Navbar/index.js (hunk 15 above)
Comment (legendarykamal) [reply site/src/sections/Navbar/index.js:15]: aligned it :+1:

File: site/src/pages/index.js (hunk 14 above)
Comment (legendarykamal) [reply site/src/pages/index.js:25]: for dev purpose now added it back

**Comment by l5io** [comment]
🚀 Preview for commit e3af6319ade3d619ea924c102c259f10f0cfff85 at: https://6316309f9c4ac3741872c662--service-mesh-patterns.netlify.app

**Comment by legendarykamal** [comment]
Now a Hamberger icon also appears at < 760px screen.

**Comment by legendarykamal** [comment]
@debo19 I have dropped in a hamberger menu too + modified the links considering they will be used to navigate around the page as per different sections , 
lets get this merged first and then we can tackle a little here and there responsiveness issues . :smiley:

**Comment by l5io** [comment]
🚀 Preview for commit 7cafc017df21bf39b50ff94541964cc53feaa568 at: https://631657e64f16cf2b280a11d6--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 79ca0fbbbd5c10bf65af7906657505b2f8d83b75 at: https://6316b3ea4f16cf68850a12ff--service-mesh-patterns.netlify.app

**Comment by Nikhil-Ladha** [comment]
A few changes still needed before I would consider merging this:
1. The nav item's color need to be reversed, i.e, by default they should be `keppel green` and on hovering the `Caribbean grean` (current color).
2. Move the two ends closer to each other, similar to the navbar structure of layer5.io site.
3. Please fix that button's default styling. Add a `Button` component, copy paste (some modification might be needed, do that) from layer5.io repo and use that.
4. The navbar is not responsive from 1180px to 768px.
5. The dropdown menu is not working as expected in mobile view. Which makes me think did you even build the site locally before pushing the changes?
6. No need to use the styled-components for each element, we can simply apply styles to the element's directly. Revert that.

**Comment by legendarykamal** [comment]
Ok @Nikhil-Ladha accepted will do the necc by today 🙂

**Comment by legendarykamal** [comment]
responsiveness fixed , just minor tweaks left

**Comment by l5io** [comment]
🚀 Preview for commit 0ff12ca81c60c98b4a08dbe7d6cef4466d201015 at: https://631b9a735fa5951b0bff8e30--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit dea57c1eb21105b49ff03cf5d5103bd5b13ff3b9 at: https://631cc469d1b60469639d7769--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 22dd8a4dd5c6cd00dced9fd77fa8f6450f282bad at: https://631e3ee6f19e9b6df8e0b295--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit f2555541db957205fb8086dc674199edb2ee7414 at: https://631e461fbb7dd76aac026137--service-mesh-patterns.netlify.app

**Comment by Nikhil-Ladha** [comment]
There is a gap between the navbar drawer and the navbar in the mobile view.
Screenshot from 2022-09-12 10-38-49

**Comment by legendarykamal** [comment]
@Nikhil-Ladha fix was made, what else needs modification in navbar?

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
No need to break up a simple navbar code into multiple files.
Please merge them all into one component as earlier.

**Comment by Nikhil-Ladha** [comment]
Also, the build seems to be failing

**Comment by legendarykamal** [comment]
> No need to break up a simple navbar code into multiple files. Please merge them all into one component as earlier.

ok

**Comment by l5io** [comment]
🚀 Preview for commit 86a91c03a95a3964dccb72225e39f85f60bf8edb at: https://63242f402139ab7fc08d382e--service-mesh-patterns.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit e4c44961d45212130a8d101bac7e8900c7f4f425 at: https://63249deac2a8216ca1629a96--service-mesh-patterns.netlify.app

File: site/src/sections/Navbar/index.js [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+        </Navbox>
+      ) : (
+        <Navbox open>
+          <NavItem to="/">Book Chapters</NavItem>
+          <NavItem to="/">Meet the Authors</NavItem>
+          <NavItem to="/">Patterns in action</NavItem>
+          <NavItem to="/about">About <img className="layer5" src={Layer5} alt="Layer5 logo" /></NavItem>
+          <NavItem to="/get"><button><span className="label">Get it Now</span></button></NavItem>
+        </Navbox>
+      )}
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/index.js:38]: I don't think you need to duplicate the whole code for the mobile menu, just using a class when `navbarOpen` is true, would do. The same applies to the `Hamburger`, if the `navbarOpen` is false just return ''

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
For the nth time, please remove that border from the button 🙏🏼 
In fact, simply fetch the latest master and rebase the branch. Then you will have the `Button` component which you can use in this component too. That would fix the button CSS.

**Comment by l5io** [comment]
🚀 Preview for commit 01c89a292bd1d962743dbd25fb3a928e282e3987 at: https://6327d91c3cd2e037006de810--service-mesh-patterns.netlify.app

File: site/src/sections/Navbar/index.js [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+          <NavItem to="/about">About <img className="layer5" src={Layer5} alt="Layer5 logo" /></NavItem>
+          <NavItem to="/get"><Button><span className="label">Get it Now</span></Button></NavItem>
+        </Navbox>
+      ) : (
+        <Navbox open>
+          <NavItem to="/">Book Chapters</NavItem>
+          <NavItem to="/">Meet the Authors</NavItem>
+          <NavItem to="/">Patterns in action</NavItem>
+          <NavItem to="/about">About <img className="layer5" src={Layer5} alt="Layer5 logo" /></NavItem>
+          <NavItem to="/get"><Button><span className="label">Get it Now</span></Button></NavItem>
```
Comment (Nikhil-Ladha) [inline site/src/sections/Navbar/index.js:36]: I don't think the CSS for the `button` and the extra `span` is required anymore.
The `Button` component provides them all. Also, for the button text use the `title` prop that we have in the `Button` component. For reference either check the component code or layer5.io repo

Comment (legendarykamal) [reply site/src/sections/Navbar/index.js:36]: I have removed the ```span``` but kept the custom css as the default button css are not optimum 

image

**Comment by l5io** [comment]
🚀 Preview for commit 0ba1104348bb8bf2a8a5505d42f35fb9cec83b00 at: https://632833b5095227007927be60--service-mesh-patterns.netlify.app

File: site/src/sections/Navbar/index.js (hunk 17 above)
Comment (Nikhil-Ladha) [reply site/src/sections/Navbar/index.js:36]: Well, you need to provide the `secondary` prop 🤷🏼‍♂️

**Comment by l5io** [comment]
🚀 Preview for commit 41bf6f5719c026fcab886b60e93c8874e3876175 at: https://632da222800fd4033dddcb11--service-mesh-patterns.netlify.app

**Comment by legendarykamal** [comment]
// @Nikhil-Ladha all suggestions have been implemented , anything else needed ? :)

**Pull request merged**